src/java.base/share/classes/jdk/internal/access/JavaLangAccess.java

```diff
@@ -391,6 +391,13 @@ public interface JavaLangAccess {
      */
     void inflateBytesToChars(byte[] src, int srcOff, char[] dst, int dstOff, int len);
 
+    boolean isLatin1(String s);
+
+    /**
+     * if string#coder() is Latin1 return the count of string#value() leading greater than zero, else return 0
```

Comment:
Can you move this next to `countPositives`? 

```suggestion
     * Count the number of leading positive, non-zero bytes in the range.
```

Technically this new routine is the "real" `countPositives` since, mathematically speaking, zero is neither positive nor negative. It's named like it is because I'm no mathematician, and now it might be better to rename away from that to disambiguate.. If you feel like it then please file an RFE to have `countPositives` renamed to `countNonNegatives`

Reply:
We might rename these to `countAscii` and `countModifiedUtf8Compatible`.

Reply:
ascii includes '\0', `CountModifiedUtf8Compatible` Newbies don't seem to know what it means

Reply:
I think naming internal low-level methods based precisely on what they do rather than their intended purpose is fine: we're counting positives and zeros in a byte array. Good! Higher level API methods called `isAscii(byte[] utf8Bytes)` et.c. which calls the low-level methods? Also good!